packages/xo-server/src/xo-mixins/authorization.mjs

```
@@ -0,0 +1,70 @@
+import { createLogger } from '@xen-orchestra/log'
+import get from 'lodash/get.js'
+
+const log = createLogger('xo:store')
+
+class UnauthorizedFeatureError extends Error {
+  constructor(featureCode) {
+    super(`feature ${featureCode} is not authorized for current installation`)
+    this.name = 'UnauthorizedFeatureError'
+    this.featureCode = featureCode
+  }
+}
+
+const FREE = 1
+const STARTER = 2
+const ENTREPRISE = 3
+const PREMIUM = 4
+const OPEN = 5
+
+const TRIAL_LEVELS = {
+  FREE,
+  STARTER,
+  ENTREPRISE,
+  PREMIUM,
+  OPEN,
+}
+
+const AUTHORIZATIONS = {
+  BACKUP: {
+    FULL: STARTER,
+    METADATA: ENTREPRISE,
+    S3: STARTER,
+  },
+}
+
+export default class Authorization {
+  #app
+  constructor(app) {
+    this.#app = app
+  }
+
+  #getMinPlan(featureCode) {
+    return get(AUTHORIZATIONS, featureCode, 0)
```

We should throw if the authorization is not found, it's a programmer error.